### DISCUSSION TITLE
[0.3.1] Remote graph not working by default

### DISCUSSION THREAD

**Discussion by darwinyip** [JanusGraph/janusgraph#1499] [Q&A]
Running `gremlin-server.sh` and `gremlin.sh` leaving all default settings from the release archive does not work when trying to connect with remote graph.

```
> ./bin/gremlin.sh
gremlin> cluster = Cluster.open()
==>localhost/127.0.0.1:8182
gremlin> client = cluster.connect()
==>org.apache.tinkerpop.gremlin.driver.Client$ClusteredClient@242b6e1a
gremlin> aliased = client.alias("g")
==>org.apache.tinkerpop.gremlin.driver.Client$AliasClusteredClient@6f94fb9d
gremlin> g = org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph.instance().traversal()
==>graphtraversalsource[emptygraph[empty], standard]
gremlin> aliased.submit(g.V())
org.apache.tinkerpop.gremlin.driver.exception.ResponseException: The traversal source [g] for alias [g] is not configured on the server.
```

**Comment by ugurcoskn** [comment]
Hey @darwinyip,

Firstly, I should say, you need to share gremlin-server logs for understand whats happening.

But in my option maybe your gremlin groovy scripts doesn’t work correctly because of your storage-backend or storage-index-backend doesnt work properly or working different ports and doesnt match janusgraph default ports.

First you need to check them and your gremlin server run at 8182.

Janusgraph using ‘**elasticsearch**’ as index backend and ‘**cql**’ storage backend with its default configuration. In this reason you should up elasticsearch before running your gremlin-server. If you use the run elasticsearch under janusgraph-0.3.1 release file, elasticsearch up at localhost:9200 (as default elasticsearch server). 

jansugraph-cql-es-server.properties:

```
gremlin.graph=org.janusgraph.core.JanusGraphFactory
storage.backend=cql
storage.hostname=127.0.0.1
storage.cql.keyspace=janusgraph
cache.db-cache = true
cache.db-cache-clean-wait = 20
cache.db-cache-time = 180000
cache.db-cache-size = 0.25
index.search.backend=elasticsearch
index.search.hostname=127.0.0.1:9200
index.search.elasticsearch.client-only=true
```


I tested it with default '**janusgraph-berkeleyje-es-server.properties**' and i get an error. Maybe default configuration should be update for berkeleyje-es.properties

I change this:
`index.search.hostname=elasticsearch`
To:
`index.search.hostname=127.0.0.1:9200`

Then i follow your steps and get remote connection without errors.

```
gremlin> cluster = Cluster.open()
==>localhost/127.0.0.1:8182
gremlin> client = cluster.connect()
==>org.apache.tinkerpop.gremlin.driver.Client$ClusteredClient@672a1c62
gremlin> aliased = client.alias("g")
==>org.apache.tinkerpop.gremlin.driver.Client$AliasClusteredClient@5edc70ed
gremlin> g = org.apache.tinkerpop.gremlin.structure.util.empty.EmptyGraph.instance().traversal()
==>graphtraversalsource[emptygraph[empty], standard]
gremlin> aliased.submit(g.V())
gremlin>
```
